include/envoy/server/guarddog.h

```diff
@@ -1,11 +1,18 @@
 #pragma once
 
 #include "envoy/common/pure.h"
+#include "envoy/common/time.h"
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
 #include "envoy/server/watchdog.h"
+#include "envoy/thread/thread.h"
 
 namespace Envoy {
 namespace Server {
 
+using GuardDogActionCb = ::std::function<void(
+    envoy::config::bootstrap::v3::Watchdog::WatchdogAction::WatchdogEvent,
+    ::std::vector<::std::pair<Thread::ThreadId, MonotonicTime>>, MonotonicTime)>;
```

Comment:
Can you document what each of these function arguments mean?

I assume that the last argument is the current time.  I think that the most interesting arguments here are the stuck thread ids and the current time.

Reply:
Done.